addons/account/models/chart_template.py

```diff
@@ -1310,3 +1325,31 @@ def _load_translations(self, langs=None, companies=None, template_data=None):
                             break
 
         translation_importer.save(overwrite=False)
+
+    @api.model
+    def _update_reconciliation_models(self, company):
+        """
+        The write-off button default reconciliation model for Bill creation
+        requires tax ids, which need to be set after the default taxes are defined.
+        """
+        lines = self.env['account.reconcile.model'].search([
+            *self.env['account.reconcile.model']._check_company_domain(company),
+            ('counterpart_type', '=', 'purchase'),
+        ]).line_ids
+
+        tax_ids = (
+            self.env['account.tax'].search([
+                *self.env['account.tax']._check_company_domain(company),
+                ('type_tax_use', '=', 'purchase'),
+                ('id', 'in', company.account_purchase_tax_id.ids),
+            ], limit=1)
+            or self.env['account.tax'].search([
+                *self.env['account.tax']._check_company_domain(company),
+                ('type_tax_use', '=', 'purchase'),
+            ], limit=1)
+            or self.env['account.tax'].search([
+                *self.env['account.tax']._check_company_domain(company),
+            ], limit=1)
```

Comment:
At this point, I think `account_sale_tax_id` / `account_purchase_tax_id` are set on the company. Use it instead?
I know @william-andre made something on the chart template to manage the circular dependencies between the models to be loaded (like account & tax). Maybe there is a better way to do that?
In any case, searching on all reconcile models and assigning taxes on all of them is bad.
Instead of doing what, what about making `tax_ids` computed on `account.reconcile.model.line` and put by default the tax from the company? That way, you can just remove the whole diff here

Reply:
What can be done, is:
* before `_get_chart_template_data` but before `_load_data` (maybe around `_pre_load_data`?
* do the same thing as here, but on `data['account.tax'] in order to avoid doing more searches. Just set the xmlid instead of the id in a `Command.set`